Doc/whatsnew/3.10.rst

```diff
@@ -225,6 +225,238 @@ See :class:`typing.Callable`, :class:`typing.ParamSpec`,
 
 (Contributed by Ken Jin in :issue:`41559`.)
 
+PEP 634: Structural Pattern Matching
+------------------------------------
+
+Structural pattern matching has been added in the form of a *match statement*,
+a data object (sequences, mappings, primitive data types as well as class
+instances) and *case statements* of patterns with associated actions. Pattern
+matching enables programs to extract information from complex data types,
+branch on the structure of data, and apply specific actions based on different
+forms of data.
+
+Syntax and operations
+~~~~~~~~~~~~~~~~~~~~~
+
+The generic syntax of pattern matching is::
+
+    def function(subject):
+        match {subject}:
+            case {pattern_1}:
+                {action_1}
+            case {pattern_2}:
+                {action_2}
+            case {pattern_3}:
+                {action_3}
+            case _:
+                {action_wildcard}
+
+A match statement takes an expression and compares its value to successive
+patterns given as one or more case blocks.  Specifically, pattern matching
+operates by::
+
+    1. taking a data object ``subject`` with type and shape
+    2. substituting the ``subject`` into a ``match`` statement
+    3. comparing the match subject with each pattern in a ``case`` statement
+       from top to bottom until a match is confirmed. If an exact match is not
+       confirmed, the last case, a wildcard ``_``, will be used as the matching
+       case.
```

Comment:
Hm, this suggests that there should always be a wildcard case at the end. Unless you are talking specifically about the above example, but in that case I would open with "This match statement takes" instead of "A match statement takes". Alternatively, you could perhaps add something like "If no wildcard pattern is present, and an exact match is not confirmed, the entire match block is a no-op." (Though maybe the whole thing about the wildcard can be moved down to a new item 5.)